app/src/ui/app.tsx

```diff
@@ -1300,6 +1313,8 @@ export class App extends React.Component<IAppProps, IAppState> {
       return null
     }
 
+    const onDismissed = this.getOnPopupDismissedFn(popup.type)
+
```

Comment:
As discussed offline with @niik, we're creating a memoized version of `onPopupDismissed` that passes the `popupType` that gets opened, so when dismissing a modal it will only close itself and not other modals that could have got opened afterwards.

For now I've decided to only use this method on the new modal. It would be good to use it for every other modal but that would require extensive testing since it's possible that some modals are relying on the fact that currently `onDismissed` closes any other modal.